pkg/storage/cacher.go

```diff
@@ -268,16 +326,23 @@ func (c *Cacher) Watch(ctx context.Context, key string, resourceVersion string,
 		return newErrWatcher(err), nil
 	}
 
+	indexValue, indexSupported := "", false
+	if c.indexFunc.IndexName != "" {
```

Comment:
nit: use len(string) == 0 just because we do that in a lot of core code as a convention